### DISCUSSION TITLE
TUYA device BN-LINK BNC-60/U153T two switch plug uses TUYA WB2S chip

### DISCUSSION THREAD

**Discussion by varocketry** [ct-Open-Source/tuya-convert#977] [Q&A]
TUYA device BN-LINK BNC-60/U153T two switch plug uses TUYA WB2S chip

Nope, no ESP Chip that I can find....   I guess this cannot be flashed to HAA/ Homekit???

https://developer.tuya.com/en/docs/iot/wb2s-module-datasheet?id=K9ghecl7kc479

![bn-link5](https://user-images.githubusercontent.com/13277352/129395641-7e59d177-cede-459f-b768-a1cf7c5ee8e6.png)
![bn-link4](https://user-images.githubusercontent.com/13277352/129395643-a186f825-b167-453c-a3a6-dd5d56da5f3c.png)
![bn-link3](https://user-images.githubusercontent.com/13277352/129395647-0929f34b-3011-4e5e-8170-04d5413e7a86.png)
![bn-link2](https://user-images.githubusercontent.com/13277352/129395651-c9842875-e928-4d5a-a979-0d416c8c27b8.png)
![bn-link1](https://user-images.githubusercontent.com/13277352/129395785-129b025d-a4a9-4aca-b08e-4696d684d5f6.png)

**Comment by mkaatman** [comment]
A project has been started to work on an open source firmware. https://www.elektroda.com/rtvforum/topic3850712.html

You may be able to swap that WB2S for an ESP12 chip. https://github.com/lsr1/wb2s-to-esp12
